spec/rules_spec.rb

```diff
@@ -44,6 +45,23 @@
     expect(EppoClient.find_matching_rule({ 'age' => '99' }, [numeric_rule])).to be_nil
   end
 
+  it 'tests find matching rule for semver string' do
+    semver_greater_than_condition = EppoClient::Condition.new(
+      operator: EppoClient::OperatorType::GTE, value: '1.0.0', attribute: 'version'
+    )
+    semver_less_than_condition = EppoClient::Condition.new(
+      operator: EppoClient::OperatorType::LTE, value: '2.0.0', attribute: 'version'
+    )
+    semver_rule = EppoClient::Rule.new(
+      allocation_key: 'allocation',
+      conditions: [semver_less_than_condition, semver_greater_than_condition]
+    )
```

Comment:
Based on our experience with other SDKs we'll need more robust test cases. For example, for the semver comparison we want to make sure it wouldn't pass if evaluated lexographically or partially parsed as a float *(e.g, ensuring it realises 1.12.0 > 1.5.0):
```rb
semver_greater_than_condition = EppoClient::Condition.new(
  operator: EppoClient::OperatorType::GTE, value: '1.2.3', attribute: 'version'
)
semver_less_than_condition = EppoClient::Condition.new(
  operator: EppoClient::OperatorType::LTE, value: '1.15.0', attribute: 'version'
)
semver_rule = EppoClient::Rule.new(
      allocation_key: 'allocation',
      conditions: [semver_less_than_condition, semver_greater_than_condition]
    )
    
    expect(EppoClient.find_matching_rule({ 'version' => '1.12.0' }, [semver_rule])).to be(semver_rule)
```

Reply:
Great point. I added that test.